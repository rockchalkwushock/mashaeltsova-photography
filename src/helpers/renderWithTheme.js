import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
