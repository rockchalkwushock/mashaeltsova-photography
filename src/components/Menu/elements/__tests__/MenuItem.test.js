import React from 'react'

import { renderWithTheme } from '../../../../helpers'
import { MenuItem } from '../index'

describe('Styled Component: <MenuItem />', () => {
  let wrapper
  let wrapper2
  beforeAll(() => {
    wrapper = renderWithTheme(<MenuItem />)
    wrapper2 = renderWithTheme(<MenuItem state />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper2).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('opacity', '0')
    expect(wrapper).toHaveStyleRule('pointer-events', 'none')
    expect(wrapper).toHaveStyleRule(
      'transition',
      'opacity 0.3s ease-in-out 0.1s'
    )
  })
  test('should have toggled styles', () => {
    expect(wrapper2).toHaveStyleRule('opacity', '1')
  })
})
