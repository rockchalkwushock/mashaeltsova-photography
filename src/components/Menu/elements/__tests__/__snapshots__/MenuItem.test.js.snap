// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <MenuItem /> should render markup 1`] = `
.c0 {
  opacity: 0;
  pointer-events: none;
  -webkit-transition: opacity 0.3s ease-in-out 0.1s;
  transition: opacity 0.3s ease-in-out 0.1s;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <MenuItem>
    <li
      className="c0"
    />
  </MenuItem>
</ThemeProvider>
`;

exports[`Styled Component: <MenuItem /> should render markup 2`] = `
.c0 {
  opacity: 1;
  pointer-events: none;
  -webkit-transition: opacity 0.3s ease-in-out 0.1s;
  transition: opacity 0.3s ease-in-out 0.1s;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <MenuItem
    state={true}
  >
    <li
      className="c0"
    />
  </MenuItem>
</ThemeProvider>
`;
